parboiled-scala/src/main/scala/org/parboiled/scala/rules/ReductionRule.scala

```diff
@@ -26,35 +27,1034 @@ abstract class ReductionRule extends Rule
 /**
  * A rule taking one value off the value stack and replacing it with another value.
  */
-class ReductionRule1[-Z, +R](val matcher: Matcher) extends ReductionRule {
-  def ~[X, Y](other: PopRule3[X, Y, R]) = new PopRule3[X, Y, Z](append(other))
-  def ~[Y](other: PopRule2[Y, R]) = new PopRule2[Y, Z](append(other))
-  def ~(other: PopRule1[R]) = new PopRule1[Z](append(other))
-  def ~[X, Y, T](other: ReductionRule3[X, Y, R, T]) = new ReductionRule3[X, Y, Z, T](append(other))
-  def ~[Y, T](other: ReductionRule2[Y, R, T]) = new ReductionRule2[Y, Z, T](append(other))
-  def ~[T](other: ReductionRule1[R, T]) = new ReductionRule1[Z, T](append(other))
-  def |[ZZ <: Z, RR >: R](other: ReductionRule1[ZZ, RR]) = new ReductionRule1[ZZ, RR](appendChoice(other))
-  protected def withMatcher(matcher: Matcher) = new ReductionRule1[Z, R](matcher).asInstanceOf[this.type]
+class ReductionRule1[-Z, +A](val matcher: Matcher) extends ReductionRule {
```

Comment:
Holy shit! :)
I hope you auto-generated all these methods....
Did you?

Reply:
If vim macros count, yes ;)

Reply:
Awesome! :)